src/hardware/setup.rs

```diff
@@ -761,10 +767,7 @@ pub fn setup(
             data
         };
 
-        let mut stack = smoltcp_nal::NetworkStack::new(
-            interface,
-            system_timer::SystemTimer::default(),
-        );
+        let mut stack = smoltcp_nal::NetworkStack::new(interface, clock);
```

Comment:
I'm not a fan of constructing the clock just to pass it in here. Could we construct the clock locally?

I dislike having to construct the `clock` externally and passing it in - it's just a strange API.

```rust
let clock = SystemTimer::new(|| crate::app::monotonics::now().ticks() as u32);
```
Not sure if that will work though.

Reply:
Doesn't work because the binaries are not part of the library crate.